gnmidiff/setrequest.go

```diff
@@ -17,13 +17,183 @@ package gnmidiff
 
 import (
 	"fmt"
+	"sort"
+	"strconv"
+	"strings"
 
 	"github.com/derekparker/trie"
 	"github.com/openconfig/ygot/ygot"
+	"github.com/openconfig/ygot/ytypes"
 
 	gpb "github.com/openconfig/gnmi/proto/gnmi"
 )
 
+// MismatchedUpdate represents two different update values for the same leaf
+// node.
+type MismatchedUpdate struct {
+	// A is the update value in A.
+	A interface{}
+	// B is the update value in B.
+	B interface{}
+}
+
+// SetRequestIntentDiff contains the intent difference between two SetRequests.
+//
+// - The key of the maps is the string representation of a gpb.Path constructed
+// by ygot.PathToString.
+// - The value of the update fields is the JSON_IETF representation of the
+// value.
+type SetRequestIntentDiff struct {
+	AOnlyDeletes      map[string]struct{}
+	BOnlyDeletes      map[string]struct{}
+	CommonDeletes     map[string]struct{}
+	AOnlyUpdates      map[string]interface{}
+	BOnlyUpdates      map[string]interface{}
+	CommonUpdates     map[string]interface{}
+	MismatchedUpdates map[string]MismatchedUpdate
+}
+
+// Format is the string format of any gNMI diff utility.
```

Comment:
"any gNMI diff utility"?  Do you really mean "any" here?

Reply:
Added "in this package." The utilities I can imagine (Set vs. Set, Set vs. Get, Get vs. Get) would all use this.